src/main/java/com/vaadin/guice/server/NavigatorManager.java

```diff
@@ -61,14 +59,14 @@ void addNavigator(UI instance) {
                 defaultViewField.getName()
         );
 
-        Navigator navigator;
+        GuiceNavigator navigator = guiceVaadin.assemble(GuiceNavigator.class);
```

Comment:
this is the line I was referring to in the comment. You would need to add another configuration option in order to tell guice which class is the actual GuiceNavigator to use. And then you can just remove the GuiceNavigator class altogether, you just need a parameter like Class<? extends Navigator> navigatorToUse that defaults to the standard com.vaadin.navigator.Navigator.